plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt

```diff
@@ -92,15 +89,33 @@ class AmazonQLanguageClientImpl(private val project: Project) : AmazonQLanguageC
     }
 
     override fun logMessage(message: MessageParams) {
-        showMessage(message)
+        val type = when (message.type) {
+            MessageType.Error -> Level.ERROR
+            MessageType.Warning -> Level.WARN
+            MessageType.Info, MessageType.Log -> Level.INFO
+        }
+
+        if (type == Level.ERROR &&
+            message.message.lineSequence().firstOrNull()?.contains("NOTE: The AWS SDK for JavaScript (v2) is in maintenance mode.") == true
+        ) {
+            LOG.info { "Suppressed Flare AWS JS SDK v2 EoL error message" }
+            return
+        }
+
+        LOG.atLevel(type).log(message.message)
     }
 
-    override fun showDocument(params: ShowDocumentParams?): CompletableFuture<ShowDocumentResult> {
+    override fun showDocument(params: ShowDocumentParams): CompletableFuture<ShowDocumentResult> {
         try {
-            if (params == null || params.uri.isNullOrEmpty()) {
+            if (params.uri.isNullOrEmpty()) {
                 return CompletableFuture.completedFuture(ShowDocumentResult(false))
             }
 
+            if (params.external == true) {
+                BrowserUtil.open(params.uri)
```

Comment:
does this need to be on edt?

Reply:
no but i dont want to rewrite this and it's not a slow operation